TunnelKit/Sources/Protocols/OpenVPN/AppExtension/OpenVPNTunnelProvider.swift

```diff
@@ -665,18 +668,26 @@ extension OpenVPNTunnelProvider: OpenVPNSessionDelegate {
         var proxySettings: NEProxySettings?
         if let httpsProxy = cfg.sessionConfiguration.httpsProxy ?? options.httpsProxy {
             proxySettings = NEProxySettings()
-            proxySettings?.httpsServer = httpsProxy.neProxy()
-            proxySettings?.httpsEnabled = true
+            proxySettings!.httpsServer = httpsProxy.neProxy()
+            proxySettings!.httpsEnabled = true
             log.info("Routing: Setting HTTPS proxy \(httpsProxy.address.maskedDescription):\(httpsProxy.port)")
         }
         if let httpProxy = cfg.sessionConfiguration.httpProxy ?? options.httpProxy {
             if proxySettings == nil {
                 proxySettings = NEProxySettings()
             }
-            proxySettings?.httpServer = httpProxy.neProxy()
-            proxySettings?.httpEnabled = true
+            proxySettings!.httpServer = httpProxy.neProxy()
+            proxySettings!.httpEnabled = true
             log.info("Routing: Setting HTTP proxy \(httpProxy.address.maskedDescription):\(httpProxy.port)")
         }
+        if let pacURL = cfg.sessionConfiguration.proxyAutoConfURL ?? options.proxyAutoConfURL {
+            if proxySettings == nil {
+                proxySettings = NEProxySettings()
+            }
+            proxySettings!.proxyAutoConfigurationURL = pacURL
+            proxySettings!.autoProxyConfigurationEnabled = true
+            log.info("Routing: Setting PAC \(pacURL)")
+        }
```

Comment:
While I understand that the result is 100% equal, I like to cut unwrapped optionals to the bare minimum. Seeing an "!" automatically rings a bell and has me wonder "is it safe to use here?", even if this is an obviously safe place to use it. So I'd rather be "visually safe" too by using the "?" instead.

Could you therefore use `proxySettings?` vs `proxySettings!` here?

Reply:
I like to use `!` in places where it's obvious it won't error/or add a comment explaining why it's ok and use it as an assert. I'll change it back to `?` to match your style.